lib/remap-functions/src/del.rs

```diff
@@ -58,15 +69,58 @@ impl Expression for DelFn {
 #[cfg(test)]
 mod tests {
     use super::*;
+    use crate::map;
 
-    test_type_def![static_type_def {
-        expr: |_| DelFn {
-            paths: vec![Path::from("foo")]
-        },
-        def: TypeDef {
-            fallible: true,
-            kind: value::Kind::Null,
-            ..Default::default()
-        },
-    }];
+    #[test]
+    fn del() {
+        let cases = vec![
+            (
+                // String field exists
+                map!["exists": "value"],
+                Ok(value!("value")),
+                DelFn::new(Path::from("exists")),
+            ),
+            (
+                // String field doesn't exist
+                map!["exists": "value"],
+                Ok(value!(null)),
+                DelFn::new(Path::from("does_not_exist")),
+            ),
+            (
+                // Array field exists
+                map!["exists": value!([1, 2, 3])],
+                Ok(value!([1, 2, 3])),
+                DelFn::new(Path::from("exists")),
+            ),
+            (
+                // Null field exists
+                map!["exists": value!(null)],
+                Ok(value!(null)),
+                DelFn::new(Path::from("exists")),
+            ),
+            (
+                // Map field exists
+                map!["exists": map!["foo": "bar"]],
+                Ok(value!(map!["foo": "bar"])),
+                DelFn::new(Path::from("exists")),
+            ),
+            (
+                // Integer field exists
+                map!["exists": 127],
```

Comment:
Would it make sense to add a test for arrays here too? Like `del(.exists[0])`? It seems like `del()` supports this looking at the below `object_remove` tests. I'm not sure where the line is between what should be tested where though.

Reply:
I can add it, seems like something we want to make sure keeps working 👍